network/setup-ip

```diff
@@ -24,6 +24,22 @@ configure_network() {
     local gateway6="$8"
     local primary_dns="$9"
     local secondary_dns="${10}"
+    local i
+
+    for i in all "$INTERFACE"; do
```

Comment:
Those options should be harmless but I'm not totally sure in the case where `$INTERFACE` serves a full separate network in the case of bridge setup. In all the case and in order to not "brick" any production system, I would propose to enable those sysctl options only if we request the specific qube to have the tag for example: `net.hardening`. Across all of our configurations, we may introduce hardening features as tag like `{net,system,etc.}.hardening`.

Reply:
That is a good idea, especially if combined with the qubesdb fixes (https://github.com/QubesOS/qubes-core-qubesdb/pull/44).

Reply:
I also wonder if this should not apply to *only* "$INTERFACE" because *all* is already for all existing interfaces. I'm not sure how this would behave for example with docker, I would need to try.

Reply:
I believe this is sysctl-specific:

- `accept_seg6` and `rpl_seg_enabled` should both be set to 0 on the interface, and doing so is sufficient.
- `ipv6.conf.$i.accept_source_route` should be set to -1 (_not_ 0!), and doing so is sufficient.
- global sysctls `net.ipv6.max_hbh_opts_length` and `net.ipv6.max_dst_opts_length` should be set to 0 to disable IPv6 options (rarely used and non-zero attack surface)
- `ipv6.conf.$i.accept_redirects` should be 0, and this is sufficient.
- `ipv6.conf.$i.accept_ra` should be 0, and this is sufficient.
- `ipv6.conf.$i.accept_dad` should be 0, and this is *not* sufficient: `ipv6.conf.all.accept_dad` must also be 0.
- `ipv4.conf.$i.accept_source_route` should be 0, and this is sufficient.
- `ipv4.conf.$i.accept_redirects` should be 0 and this is *not* sufficient: `ipv4.conf.all.accept_redirects` must also be 0.
- It *might* be possible to use a sysctl file for this, but that is _not_ tested.  It would cover all `eth*` and `vif*` interfaces.